SECP-service/src/test/java/com/visucius/secp/Controllers/UserRegistrationControllerTest.java

```diff
@@ -61,13 +63,32 @@ public void LastNameIsTooLongTest() {
         assertEquals(response.getMessage(), UserErrorMessage.USER_NOT_CREATED);
     }
 
+    @Test
+    public void DisplayNameIsTooLongTest()
+    {
+        UserRegistrationRequest request = new UserRegistrationRequest(
+            "verrylongfirstnameamefdsafdsafsdfddfdddddsssssssssfsdfsfsdfsfsdfsdfsdffdsfsfsfsf",
+            "farah",
+            "alifarah",
+            "verylongdisplaynameaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+            "test@gmail.com",
+            "Password1");
+        UserRegistrationResponse response = controller.registerUser(request);
+
+        assertFalse(response.success);
```

Comment:
Again this is just personal preference but I feel like good unit tests only have 1 assert so you can pinpoint which test failed and that you dont have to redo asserts. So you could have multiple tests for a bad request, another test for a display name invalid, and another test for a user not created all because the username is too long, and that way you dont have to keep asserting bad request, displayname invalid and user not created over and over again and you could see when it fails rather than digging through. But it doesnt really matter since they pass and do what was expected. TLDR: I'm picky with these tests because you have multiple asserts in a single test, but I don't mind it.

Reply:
I agree, tests should be 1:1 with an assert.